File: Programmers/2202/다리를 지나는 트럭.js
function solution(bridge_length, weight, truck_weights) {
  var answer = 0;

  const queue = [];
  let bridgeWeight = 0;
  for (let i = 0; i < bridge_length; i++) {
    queue.push(0);
  }

  while (queue.length !== 0) {
    answer += 1;
    bridgeWeight -= queue.shift();

    if (truck_weights.length !== 0) {
      if (bridgeWeight + truck_weights[0] <= weight) {
        let truck = truck_weights.shift();
        queue.push(truck);
        bridgeWeight += truck;
      } else {
        queue.push(0);
      }
    }
  }

  return answer;
}

console.log(solution(2, 10, [7, 4, 5, 6]));

File: Programmers/2110/다리를 지나는 트럭.py
from collections import deque

def solution(bridge_length, weight, truck_weights):
    answer = 0
    queue = deque([0] * bridge_length) 
    truck_queue = deque(truck_weights)   
    bridge_weight = 0
    
    while queue:
        print(answer, truck_queue, bridge_weight)
        answer += 1
        bridge_weight -= queue.popleft()
        
        if truck_queue:
            if bridge_weight + truck_queue[0] <= weight:
                truck_weight = truck_queue.popleft()
                bridge_weight += truck_weight
                queue.append(truck_weight)
            else:
                queue.append(0)
                
    
    return answer

print(solution(2,10,[7,4,5,6]))